python/katana/lonestar/analytics/deg_count.py

```diff
@@ -0,0 +1,68 @@
+import numpy as np 
+import pyarrow
+
+from katana.property_graph import PropertyGraph
+from katana.datastructures import LargeArray, AllocationPolicy
+from katana.loops import do_all, do_all_operator
+from katana.atomic import (
+    atomic_add,
+)
+
+@do_all_operator()
+def initialize_in_deg(nin, nid):
+    nin[nid] = 0
+
+@do_all_operator()
+def count_in_and_out_deg(graph: PropertyGraph, nout, nin, nid):
+    outdeg = 0
+    for edge in graph.edges(nid):
+        outdeg += 1
+        dst = graph.get_edge_dst(edge)
+        atomic_add(nin, dst, 1)
+    nout[nid] = outdeg
+
+@do_all_operator()
+def count_weighted_in_and_out_deg(graph: PropertyGraph, nout, nin, weightArr, nid):
+    outdeg = 0
+    for edge in graph.edges(nid):
+        weight = weightArr[edge]
+        outdeg += weight
+        dst = graph.get_edge_dst(edge)
+        atomic_add(nin, dst, weight)
+    nout[nid] = outdeg
+
+""" 
+   Calculate the (potentially weighted) in & out degrees of a graph
+   Will add the in & out degrees as 2 new node properties of the graph
+   weight_property: an edge property to use in calculating the weighted degree
+   inProp: the property name for the in degree
+   outProp: the property name for the out degree
+"""
+def calculate_degree(graph: PropertyGraph, weight_propery=None, inProp="in_degree", outProp="out_degree"):
```

Comment:
`in_degree_property` and `out_degree_property`. Also move those two before `weight_property` so it's easier to leave unweighted while setting the output property names. I'm not sure I like having default output names. I could be talked into it, but since the user of the function will need to reference the property later, it's best for them to explicitly provide a name.

`propery` is misspelled.